mockgen/import_mode.go

```diff
@@ -0,0 +1,352 @@
+package main
+
+import (
+	"fmt"
+	"go/token"
+	"go/types"
+	"math/rand"
+	"os"
+
+	"go.uber.org/mock/mockgen/model"
+	"golang.org/x/tools/go/packages"
+)
+
+type importModeParser struct {
+	pkgName string
+}
+
+func (p *importModeParser) parsePackage(packageName string, ifaces []string) (*model.Package, error) {
+	p.pkgName = packageName
+
+	pkg, err := p.loadPackage(packageName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load package: %w", err)
+	}
+
+	interfaces, err := p.extractInterfacesFromPackage(pkg, ifaces)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract interfaces from package: %w", err)
+	}
+
+	return &model.Package{
+		Name:       pkg.Types.Name(),
+		PkgPath:    packageName,
+		Interfaces: interfaces,
+	}, nil
+}
+
+func (p *importModeParser) loadPackage(packageName string) (*packages.Package, error) {
+	const fakeFileTemplate = `
```

Comment:
I don't quite understand how this can be implemented yet.

If you mean parsing the desired package by finding it in GOPATH? Then you will have to search through the directory tree for the necessary package yourself. I don't think it will be easier.

Maybe I didn't understand you and you meant something else?

Reply:
You shouldn't need to manually find the package in `GOPATH`. As long as go environment is configured correctly, I think that `packages.Load` (which I believe calls into `go list` under the hood) should be able to find it just as well as when it's being used as an import in a synthetic program.

Just to be sure, I cloned this PR and tested locally - if I create a trivial package at `GOPATH/src/my/package/foo`, and directly call `packages.Load` with `packageName` as the passed in pattern in this function, the following all seem to work fine: 
* `mockgen my/package/foo Fooer`, 
* `mockgen io Writer`
* `go get go.uber.org/zap`
* `mockgen go.uber.org/zap/zapcore ArrayMarshaler`
* etc

Let me know if I'm missing something here.

Reply:
Yes, it seems to work that way and looks clearer. I had doubts about packages inside `vendor`, but missing in `GOPATH`. Yes, it also works correctly.